core/templates/dev/head/pages/exploration_player/ExplorationPlayerStateService.js

```diff
@@ -0,0 +1,143 @@
+        _version, returnDict.session_id, GLOBALS.collectionId);
+      if (ENABLE_PLAYTHROUGH_RECORDING) {
+        PlaythroughService.initSession(explorationId, version);
+      }
+      PlayerCorrectnessFeedbackEnabledService.init(
+        returnDict.correctness_feedback_enabled);
+      ExplorationEngineService.init(
+        returnDict.exploration, returnDict.version,
+        returnDict.preferred_audio_language_code,
+        returnDict.auto_tts_enabled,
+        arePretestsAvailable ?
+        function() {} : callback);
+    };
+
+    var _initializePretestServices = function(pretestQuestionDicts, callback) {
+      PlayerCorrectnessFeedbackEnabledService.init(true);
+      PretestEngineService.init(pretestQuestionDicts, callback);
+    };
+
+    var _setExplorationMode = function() {
+      _inPretestMode = false;
+      _currentEngineService = ExplorationEngineService;
+    };
+
+    var _setPretestMode = function() {
+      _inPretestMode = true;
+      _currentEngineService = PretestEngineService;
+    };
+
+    return {
+      getCurrentEngineService: function() {
+        return _currentEngineService;
+      },
+      isInPretestMode: function() {
+        return _inPretestMode;
+      },
+      getPretestQuestionCount: function() {
+        return PretestEngineService.getPretestQuestionCount();
+      },
+      moveToExploration: function(callback) {
+        _setExplorationMode();
+        ExplorationEngineService.moveToExploration(callback);
+      },
+      initializePlayer: function(callback) {
+        PlayerTranscriptService.init();
+        if (_editorPreviewMode) {
+          _setExplorationMode();
+          EditableExplorationBackendApiService.fetchApplyDraftExploration(
+            explorationId).then(function(returnDict) {
+            ExplorationEngineService.init(
+              returnDict, null, null, null, _initializeDirectiveComponents);
+            PlayerCorrectnessFeedbackEnabledService.init(
+              returnDict.correctness_feedback_enabled);
+            NumberAttemptsService.reset();
+          });
+        } else {
+          if (_version) {
+            $q.all([
+              ReadOnlyExplorationBackendApiService.loadExploration(
+                _explorationId, _version, _storyId),
+              PretestQuestionBackendApiService.fetchPretestQuestions(
+                _explorationId, _storyId
+              )]).then(function(returnValues) {
+              if (returnValues[1].length > 0) {
+                _setPretestMode();
+                _initializeExplorationServices(
+                  returnValues[0], true, callback);
+                _initializePretestServices(returnValues[1], callback);
+              } else {
+                _setExplorationMode();
+                _initializeExplorationServices(
+                  returnValues[0], false, callback);
+              }
+            });
+          } else {
+            $q.all([
+              ReadOnlyExplorationBackendApiService.loadExploration(
```

Comment:
Ditto. It seems like you're using the same call as above but the second arg switches between version or story ID ... which means one of them must be wrong.

Reply:
Done.